crates/ra_hir/src/code_model_api.rs

```diff
@@ -213,8 +226,39 @@ impl Enum {
         Ok(db.enum_data(self.def_id)?.name.clone())
     }
 
-    pub fn variants(&self, db: &impl HirDatabase) -> Cancelable<Vec<(Name, Arc<VariantData>)>> {
-        Ok(db.enum_data(self.def_id)?.variants.clone())
+    pub fn variant_ids(&self) -> &[DefId] {
+        &self.variants
+    }
+}
+
+#[derive(Debug, Clone, PartialEq, Eq, Hash)]
+pub struct EnumVariant {
+    pub(crate) def_id: DefId,
+    pub(crate) enum_def_id: DefId,
+}
+
+impl EnumVariant {
+    pub fn new(def_id: DefId, enum_def_id: DefId) -> Self {
+        EnumVariant {
+            def_id,
+            enum_def_id,
+        }
+    }
+
+    pub fn def_id(&self) -> DefId {
+        self.def_id
+    }
+
+    pub fn enum_def_id(&self) -> DefId {
```

Comment:
I think this should rather return the `Enum` object (which will be easy once `Enum` contains just the `DefId`)